src/format_string.c

```diff
@@ -45,7 +45,7 @@ rcutils_format_string_limit(
   va_list args2;
   va_copy(args2, args1);
   // first calculate the output string
-  size_t bytes_to_be_written = rcutils_vsnprintf(NULL, 0, format_string, args1);
+  size_t bytes_to_be_written = (size_t)rcutils_vsnprintf(NULL, 0, format_string, args1);
```

Comment:
Like in error_handling.c, the return value of `rcutils_vsnprintf` is actually `int`, not `size_t`, since it can fail.  The same comments apply about storing in an int, checking for negative, and then casting where it is used.